Core/Models/Device.cs

```diff
@@ -0,0 +1,182 @@
+﻿using Core.Utils;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using System.Xml.Serialization;
+
+namespace Core.Models
+{
+    [XmlRoot("device")]
+    public class Device
+    {
+        [XmlElement(ElementName = "name")]
+        public string Name;
+
+        [XmlElement(ElementName = "version")]
+        public string Version { get; set; }
+
+        [XmlElement(ElementName = "description")]
+        public string Description;
+
+        [XmlElement(ElementName = "cpu")]
+        public Cpu Cpu { get; set; }
+
+        [XmlElement(ElementName = "addressUnitBits")]
+        public int MinimumBitsAddressable { get; set; }
+
+        [XmlElement(ElementName = "width")]
+        public int MaximumSingleTransferDataWidthBits { get; set; }
+
+        [XmlElement(ElementName = "size")]
+        public string DefaultRegisterWidthBitsString
+        {
+            get => string.Concat("0x", DefaultRegisterWidthBits.ToString("X2"));
+            set => DefaultRegisterWidthBits = Convert.ToUInt32(value, 16);
+        }
+
+        [XmlIgnore]
+        public uint DefaultRegisterWidthBits { get; set; }
+
+        [XmlElement(ElementName = "resetValue")]
+        public string ResetValueString
+        {
+            get => string.Concat("0x", ResetValue.ToString("X8"));
+            set => ResetValue = Convert.ToUInt32(value, 16);
+        }
+
+        [XmlIgnore]
+        public uint ResetValue { get; set; }
+
+        [XmlElement(ElementName = "resetMask")]
+        public string DefaultResetMaskString
+        {
+            get => string.Concat("0x", DefaultResetMask.ToString("X8"));
+            set => DefaultResetMask = Convert.ToUInt32(value, 16);
+        }
+
+        [XmlIgnore]
+        public uint DefaultResetMask { get; set; }
```

Comment:
Because serializer cannot directly convert a value such as "0xZZZZ" to the numeric type. To solve that problem, I've used property pairs with names [PropertyName]String/[PropertyName], e.g., string ResetValueString/uint ResetValue, where the numeric type is marked with XmlIgnore attribute. [PropertyName]String is just a decorator for the [PropertyName], it's get/set pair operates with [PropertyName]